runtime/doc/lua.txt

```diff
@@ -677,17 +697,23 @@ Tree-sitter queries are supported, with some limitations. Currently, the only
 supported match predicate is `eq?` (both comparing a capture against a string
 and two captures against each other).
 
-vim.treesitter.parse_query(lang, query)
-						*vim.treesitter.parse_query(()*
-	Parse the query as a string. (If the query is in a file, the caller
+vim.treesitter.parse_query({lang}, {query})
+						*vim.treesitter.parse_query()*
+	Parse {query} as a string. (If the query is in a file, the caller
         should read the contents into a string before calling).
 
-query:iter_captures(node, bufnr, start_row, end_row)
+vim.treesitter.get_query({lang}, {query_name})
+						*vim.treesitter.get_query()*
+	Get the runtime query named {query_name}. This internally calls
+	|vim.treesitter.parse_query()|. This will "source" all the runtime and
+	user defined queries (see |lua-treesitter-files|).
```

Comment:
Not sure you need to explain what it does internally. I also don't know what you mean by "all" the runtime and user defined queries. It doesn't just source the one that I have with `{query_name}`?

Reply:
It actually will concatenate every query named `{query_name}` and parse that as the runtime query.

Reply:
sorry I asked the same question at the bottom :$ I think it's important to describe since it is not how nvim operates in general.

Reply:
Hmmm... yeah I'm not sure I like that all that much. Why do we smash them all together?

Reply:
That's the way we did it in `nvim-treesitter` and it works well for highlighting. My worry though is for other kind of queries (i.e. locals or textobjects), because you can't overwrite part of the queries, but just add some other queries.

After some thinking, I think that this is a blocker for this PR, and we might want to find a way for users to both override and add new items to the queries. This echoes the following in the traking issue `consider a standard macro/template system for developing highlight queries (could involve fennel :)`. I actually agree that fennel could be a good idea, as it will let us write something like this (I don't know if that's valid fennel)
```fennel
(let [query {}] 
  (set query.function "((function) @test)" ;; Maybe a macro can be used here to avoid the quotes
)
```

Does this suit you ?